Client/Server code

Captainyogurt100 opened the pull request:
I just about finished the client/server code. It's now at the point where we can kinda start receiving messages and processing game logic. socket/server code should have the necessary functionality to send and receive messages.

alaynaWoleslagle committed:
Add files via upload

Commit:
Refactored Socket Connection

Made ClientCommunicator and ServerCommunicator inner classes. Created separate main classes that initialize the client and server. Moved Socket creation out of main function and into respective server and client classes. Fixed some formatting issues. Added try/catch instead of throwing IOExceptions that way we can create custom Exceptions if need be. Added Server/Client classes to separate package

Commit:
Removed ClientHandler class

Removed unnecessary ClientHandler class and created Anonymous Runnable class to override its run method.

Commit:
Sending Message Objects

Sockets now send and receive serialized Objects. Client class is now a singleton class.
Created Student Object class to test socket properties.
Added Javadoc. 
Basic readability improvements.

Commit:
Incremental Update 

TL;DR - Decoupled client/server implementation from game processing.

- Created MessageReceiver class that processes incoming messages. This class acts as the interface between the socket and game logic. When messages are received over the socket, they are added to a queue with is continuously being worked on by a separate thread. There is a specific function that should serve as the  an entry point for message logic in this class.

Made Server and Client classes Singleton classes. We only need one instance of those classes. 

Handled some logic to stop threads and close program is client is disconnected or server is closed. Although, we should give client ability to reconnect.

Created a Test class for serve as the basis of testing socket and game logic.

Commit:
Added git ignore for DS_Stores

ryanlpeterson commented:
Looks good to me. We can strip out the student test code once we've got the actual game objects setup.

Commit:
Created MessageReceiver, PlayerManager, Message classes

This primary update involves creating MessageReceiver classes to process incoming Messages. Created PlayerManager class which is responsible for maintaining current player and other player objects. Started creating Message classes

Commit:
Refactored class name, Remote unnecessary message

Changed name of Player class to PlayerStatusMessage. PlayerStatusMessage will send the status of player, whether new initialization, player added/removed, or general status. Removed ClientInitMessage because it was unnecessary.

Commit:
Ignore .DS_Store

Commit:
Ignore all .class files

Commit:
Adding player start functionality.

Added logic to send and receive messages pertaining to notifications on player ready state. Added Action enum to BaseMessage. Removed Constructors from PlayerStatusMessage class. Removed Student test class. removed static functions from MessageReceiver classes. Created Player object class.
